static/js/components/PublishDrawer.tsx

```diff
@@ -123,7 +123,7 @@ const PublishingOption: React.FC<PublishingOptionProps> = (props) => {
           )}
           <PublishForm
             onSubmit={handlePublish}
-            disabled={!publishingInfo.hasUnpublishedChanges}
+            disabled={!publishingInfo.hasUnpublishedChanges || (!website.has_site_metadata && publishingEnv==="production")}
```

Comment:
Would it be better to use the `PublishingEnv` enum from `static/js/constants.ts`?

```suggestion
            disabled={!publishingInfo.hasUnpublishedChanges || (!website.has_site_metadata && publishingEnv===PublishingEnv.Production)}
```

Reply:
yes, seems more clean